### PR TITLE
rysweet-4150-xlang-ci-test

### PR DESCRIPTION
**Pull request by rysweet**
adding tests that use the aspire xlang project to do e2e ci

### CODE REVIEW

File: dotnet/test/AutoGen.Tests/MiddlewareTest.cs [hunk 1]
```diff
@@ -33,7 +33,8 @@ public async Task HumanInputMiddlewareTestAsync()
 
         var alwaysAskUserInputMW = new HumanInputMiddleware(
             mode: HumanInputMode.ALWAYS,
-            getInput: () => "input");
+            getInput: () => "input"
```
[the remark this answers is not in the document]
Comment (rysweet) [reply dotnet/test/AutoGen.Tests/MiddlewareTest.cs]: weird - I think maybe that cam from format. yes.

File: dotnet/test/Microsoft.AutoGen.Integration.Tests/Microsoft.AutoGen.Integration.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\samples\Hello\Hello.AppHost\Hello.AppHost.csproj" />
+  </ItemGroup>
+
+  <!-- Properties, Items, and targets to ensure Python apps are initialized -->
+  <PropertyGroup>
+    <PythonVirtualEnvironmentName>.venv</PythonVirtualEnvironmentName>
+    <PythonVenvRoot>..\..\..\python</PythonVenvRoot>
```
Comment (LittleLittleCloud) [inline dotnet/test/Microsoft.AutoGen.Integration.Tests/Microsoft.AutoGen.Integration.Tests.csproj:37]:
```suggestion
    <PythonVenvRoot>$(RepoRoot)..\python</PythonVenvRoot>
```

`RepoRoot` points to `./dotnet` folder

File: dotnet/test/Microsoft.AutoGen.Integration.Tests/Microsoft.AutoGen.Integration.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  </PropertyGroup>
+
+  <Target Name="CreatePythonVirtualEnvironments" AfterTargets="PrepareForBuild">
+    <PropertyGroup>
+      <VenvDir>$(PythonVenvRoot)$(PythonVirtualEnvironmentName)/</VenvDir>
```
Comment (LittleLittleCloud) [inline dotnet/test/Microsoft.AutoGen.Integration.Tests/Microsoft.AutoGen.Integration.Tests.csproj:43]: `$(PythonVenvRoot)/$(PythonVirtualEnvironmentName)/` ?

File: dotnet/test/Microsoft.AutoGen.Integration.Tests/Microsoft.AutoGen.Integration.Tests.csproj [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      <CreateVenv Condition="!Exists('$(VenvDir)')">True</CreateVenv>
+    </PropertyGroup>
+
+    <Message Importance="Normal" Text="Initializing virtual environment for $(PythonVenvRoot)" />
+
+    <Exec Command="uv sync --all-extras" WorkingDirectory="$(PythonVenvRoot)" Condition=" '$(CreateVenv)' == 'True' " />
+    <Exec Command="source .venv/bin/activate" WorkingDirectory="$(PythonVenvRoot)" Condition=" '$(CreateVenv)' == 'True' " />
```
Comment (LittleLittleCloud) [inline dotnet/test/Microsoft.AutoGen.Integration.Tests/Microsoft.AutoGen.Integration.Tests.csproj:50]: `source` is not available on windows.

File: dotnet/test/Microsoft.AutoGen.Integration.Tests/Microsoft.AutoGen.Integration.Tests.csproj (hunk 3 above)
Comment (rysweet) [reply dotnet/test/Microsoft.AutoGen.Integration.Tests/Microsoft.AutoGen.Integration.Tests.csproj]: done

Comment (rysweet) [reply dotnet/test/Microsoft.AutoGen.Integration.Tests/Microsoft.AutoGen.Integration.Tests.csproj]: done

File: dotnet/test/Microsoft.AutoGen.Integration.Tests/Microsoft.AutoGen.Integration.Tests.csproj (hunk 4 above)
Comment (rysweet) [reply dotnet/test/Microsoft.AutoGen.Integration.Tests/Microsoft.AutoGen.Integration.Tests.csproj]: fixed

File: .github/workflows/dotnet-build.yml [hunk 5]
```diff
@@ -76,12 +76,18 @@ jobs:
     - name: list available kernels
       run: |
         python -m jupyter kernelspec list
-    - name: Setup .NET
+    - uses: astral-sh/setup-uv@v3
```
Comment (LittleLittleCloud) [inline .github/workflows/dotnet-build.yml:79]: Do we still need to setup uv if not running intergration tests in this job?

File: .github/workflows/dotnet-build.yml [hunk 6]
```diff
@@ -96,7 +102,57 @@ jobs:
...
         dotnet build --no-restore --configuration Release -bl /p:SignAssembly=true
     - name: Unit Test
-      run: dotnet test --no-build -bl --configuration Release
+      run: dotnet test --no-build -bl --configuration Release --filter type=!integration
+  
+  integration-test:
+    strategy:
+      fail-fast: true 
+      matrix:
+        os: [ ubuntu-latest]
```
Comment (LittleLittleCloud) [inline .github/workflows/dotnet-build.yml:111]: In later PR we can add `win` in os matrix

File: .github/workflows/dotnet-build.yml (hunk 5 above)
Comment (rysweet) [reply .github/workflows/dotnet-build.yml:79]: yes because of the build requirements for the test

File: .github/workflows/dotnet-build.yml [hunk 7]
```diff
@@ -96,7 +102,57 @@ jobs:
...
+    - run: uv sync --locked --all-extras
+      working-directory: ./python
+    - name: Prepare python venv
+      run: |
+        source ${{ github.workspace }}/python/.venv/bin/activate
+    - name: Setup .NET 9.0
+      uses: actions/setup-dotnet@v4
+      with:
+        dotnet-version: '9.0.x'
+    - name: Install Temp Global.JSON
```
Comment (LittleLittleCloud) [inline .github/workflows/dotnet-build.yml:137]: use `dotnet new globaljson --sdk-version 9.0.101 --roll-forward latestFeature` to create temporary global.json?

File: .github/workflows/dotnet-build.yml (hunk 6 above)
Comment (rysweet) [reply .github/workflows/dotnet-build.yml:111]: https://github.com/microsoft/autogen/issues/4618

File: .github/workflows/dotnet-build.yml (hunk 7 above)
Comment (rysweet) [reply .github/workflows/dotnet-build.yml:137]: echo is fast and it works - but its good to know this capability is there.

File: dotnet/test/Microsoft.AutoGen.Integration.Tests/HelloAppHostIntegrationTests.cs [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+    {
+        var appHost = await DistributedApplicationTestFactory.CreateAsync(appHostPath, testOutput);
+        await using var app = await appHost.BuildAsync().WaitAsync(TimeSpan.FromSeconds(15));
+
+        await app.StartAsync().WaitAsync(TimeSpan.FromSeconds(120));
+        await app.WaitForResourcesAsync().WaitAsync(TimeSpan.FromSeconds(120));
+
+        app.EnsureNoErrorsLogged();
+        await app.StopAsync().WaitAsync(TimeSpan.FromSeconds(15));
+    }
```
Comment (LittleLittleCloud) [inline dotnet/test/Microsoft.AutoGen.Integration.Tests/HelloAppHostIntegrationTests.cs:23]: Can we add verification on exiting log as well after `app.StopAsync` to ensure the app terminate correctly

**Review by LittleLittleCloud — APPROVED** [review]

File: dotnet/test/Microsoft.AutoGen.Integration.Tests/HelloAppHostIntegrationTests.cs (hunk 8 above)
Comment (rysweet) [reply dotnet/test/Microsoft.AutoGen.Integration.Tests/HelloAppHostIntegrationTests.cs:23]: https://github.com/microsoft/autogen/issues/4619

**Review by ekzhu — APPROVED** [review]

**Pull request merged**